### PR TITLE
Support for running scoverage on cypher compiler

### PR DESCRIPTION
**Pull request by boggle**

### CODE REVIEW

**Comment by thobe** [comment]
It looks like this sets a threshold for the coverage in the build itself.

That would mean that in a full build, the build would stop with cypher before reaching other modules if the coverage is insufficient. I don't think this is a good model.

IMO it would be a lot better if the coverage plugin was just concerned with generating a coverage report. As long as that report is in one of the standardised coverage report formats it will be possible for the build server to analyse the coverage summary and fail the build if the coverage is insufficient, after the entire build has completed.

I think this would be much more helpful.

**Comment by thobe** [comment]
How do we use the newly added plugin to have the build server generate coverage reports?

**Comment by boggle** [comment]
It does generate a standard coverage report, we do have to run mvn scoverage:check for this though on the build

File: community/cypher/cypher-compiler-2.2/pom.xml [hunk 1]
```diff
@@ -46,14 +48,73 @@
...
+      <!-- run with mvn scoverage:check -->
+
+      <plugin>
+        <groupId>org.scoverage</groupId>
+        <artifactId>scoverage-maven-plugin</artifactId>
+        <version>${scoverage.plugin.version}</version>
+        <configuration>
+          <scalaVersion>${scala.version}</scalaVersion>
+          <minimumCoverage>68</minimumCoverage>
+          <failOnMinimumCoverage>true</failOnMinimumCoverage>
```
Comment (davidegrohmann) [inline community/cypher/cypher-compiler-2.2/pom.xml]: Remove this and rely on TC to fail the build.

**Comment by systay** [comment]
What is the status on this?

**Comment by boggle** [comment]
The PR was read for ages due to some unrelated glitch. It can go in I believe. We need to do additional build work though to have it at least being checked by TC.

**Comment by davidegrohmann** [comment]
There are 2 problems with this solution:
1) the html files are generated with the wrong urls (extra leading '/' in the path)
2) we need to parse the output and make the build fail if the test coverage is under the limit

Not sure if it is valuable to merge this in as it is.

**Comment by davidegrohmann** [comment]
Close this for now.

**Pull request closed without merging**